modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj

```diff
@@ -472,6 +488,14 @@
   [driver [_ arg power]]
   (hx/call :power (hx/cast :float (sql.qp/->honeysql driver arg)) (sql.qp/->honeysql driver power)))
 
+(defmethod sql.qp/->honeysql [:sqlserver :percentile]
+  [driver [_ arg val]]
+  (hx/raw (apply format "APPROX_PERCENTILE_CONT (%s) WITHIN GROUP (ORDER BY %s)"
```

Comment:
I believe this does not open us to any vulnerability thanks to use of `sql.qp/->honeysql`. Let me know if that's just naive and I'll further investigate.

Reply:
I don't expect `sql.qp/->honeysql` would prevent injections. (Using prepared statement parameters for input does for sure.) Tests are needed anyway, so add some and try to inject something to see if you can convince yourself and others that this is not a problem.

Reply:
I've changed the code to use `hformat/fn-handler` to extend honeysql with `::approx-percentile-cont` function. With regards to injections, I've tried various ways how to camouflage some code into parameters, unsuccessfully.